backend/fbdev.c

```diff
@@ -22,6 +22,20 @@
 #define FBDEV_DEFAULT "/dev/fb0"
 #define SCREEN(x) ((twin_context_t *) x)->screen
 #define PRIV(x) ((twin_fbdev_t *) ((twin_context_t *) x)->priv)
+#define ARGB32_TO_RGB565(pixel)                                  \
+    (((pixel & 0x00f80000) >> 8) | ((pixel & 0x0000fc00) >> 5) | \
+     ((pixel & 0x000000f8) >> 3))
+
+/* Requires validation in 24-bit per pixel environments. */
+#define ARGB32_TO_RGB888(pixel) (0xff000000 | (pixel))
+#define GET_TWIN_FBDEV(left, top, closure, dest)                               \
```

Comment:
Once we have macro `FBDEV_PUT_SPAN_IMPL` to generate `_twin_fbdev_put_span16`, `_twin_fbdev_put_span24`, and `_twin_fbdev_put_span32`, the macro `GET_TWIN_FBDEV` can be dropped.